include/cadmium/engine/pdevs_engine_helpers.hpp

```diff
@@ -377,39 +323,20 @@ namespace cadmium {
             return std::apply(check_empty, box);
         }
 
-        //priting messages
-        template<size_t s, typename... T>
-        struct print_messages_by_port_impl{
-            using current_bag=typename std::tuple_element<s-1, std::tuple<T...>>::type;
-            static void run(std::ostream& os, const std::tuple<T...>& b){
-                print_messages_by_port_impl<s-1, T...>::run(os, b);
-                os << ", ";
-                os << boost::typeindex::type_id<typename current_bag::port>().pretty_name();
-                os << ": ";
-                implode(os, cadmium::get_messages<typename current_bag::port>(b));
-            }
-        };
-
-        template<typename... T>
-        struct print_messages_by_port_impl<1, T...>{
-            using current_bag=typename std::tuple_element<0, std::tuple<T...>>::type;
-            static void run(std::ostream& os, const std::tuple<T...>& b){
-                os << boost::typeindex::type_id<typename current_bag::port>().pretty_name();
-                os << ": ";
-                implode(os, cadmium::get_messages<typename current_bag::port>(b));
-            }
-        };
-
-        template<typename... T>
-        struct print_messages_by_port_impl<0, T...>{
-            static void run(std::ostream& os, const std::tuple<T...>& b){}
-        };
+        //Checking all dynamic bag of inbox or outbox are empty
+        template<class BOX>
```

Comment:
I like the use of BOX for the collection of BAGS, it will make it more readable. :)

Reply:
Great, glad you like it. although if the tuple message bags system must be completely removed, this must be changed to, hope that won't be the case.

Reply:
But as you have said is more readable because it uses the BOX as the map iterator of the map keys.